src/diffusers/schedulers/scheduling_ddim.py

```diff
@@ -122,6 +158,13 @@ class DDIMScheduler(SchedulerMixin, ConfigMixin):
             (https://arxiv.org/abs/2205.11487). Valid only when `thresholding=True`.
         sample_max_value (`float`, default `1.0`):
             the threshold value for dynamic thresholding. Valid only when `thresholding=True`.
+        timestep_spacing (`str`, default `"leading"`):
+            The way the timesteps should be scaled. Refer to Table 2. of [Common Diffusion Noise Schedules and Sample
+            Steps are Flawed](https://arxiv.org/abs/2305.08891) for more information.
+        rescale_betas_zero_snr (`bool`, default `False`):
+            whether to rescale the betas to have zero terminal SNR (proposed by https://arxiv.org/pdf/2305.08891.pdf).
+            This can enable the model to generate very bright and dark samples instead of limiting it to samples with
+            medium brightness.
```

Comment:
Could this potentially deprecate [offset noising](https://www.crosslabs.org/blog/diffusion-with-offset-noise)?

Reply:
Guess it's worth linking to it!

Reply:
From my own testing of zero terminal + cfg rescaling I believe so.  I'm producing samples with great variety in contrast similar to offset noise.

Offset noising does not see to be stable and eventually diverges, and requires some other yet-to-be-invented control to adjust how much offset noise is used.  The blog posts value of 0.1 works only for some number of steps, and I've found 0.1 to be far too high to for much beyond a short dreambooth style training.

![gs19999-2-nayo ff7r full length shot](https://github.com/huggingface/diffusers/assets/15839076/db206ecf-ebcf-4322-8b65-fe2d556ae6b8)
Typically above would produce a grey gradient background as the model tries to paint an image with mean brightness 0.5

![gs09599-1-a blue volvo wagon made of legos minature lego model of a volvo wagon](https://github.com/huggingface/diffusers/assets/15839076/ec7b45ec-30a1-495b-afa6-894e08d70ed5)
Likewise sometimes difficult to get bright white backgrounds . 

To reproduce the above with offset noise it may take a few attempts at guessing appropriate constant to multiple the offset noise (0.01, 0.02, etc).  Zero terminal snr appears instead to simple be stable and requires no tuning.

Reply:
I think the implementation can replace offset, see section 5.3 in the paper